File: Python/Recursion/reverse_string.py
def reverse_string(s: str, index: int) -> None:
    if index < 0:
        return
    else:
        print(s[index], end='')
        reverse_string(s, index-1)
        
def string_input() -> None:
    s = input("Enter the string: ")
    reverse_string(s, len(s)-1)

if __name__ == '__main__':
    string_input()

File: C++/Recursion/reverse_string.cpp
#include <iostream>

void reverse_string(std::string s, int index);

int main() {

    std::string s;
    std::cout << "Please enter the string : ";
    std::getline(std::cin,s);
    reverse_string(s, s.length());
    return 0;
}

void reverse_string(std::string s, int index) {

    if(index < 0) return;
    
    else {
        std::cout << s[index-1];
        reverse_string(s, index-1);
    }

}